src/snapshots/fetch.h

```
@@ -116,16 +116,49 @@ namespace snapshots
     HeaderMap headers;
   };
 
-  static inline SimpleHTTPResponse make_curl_request(
-    const SimpleHTTPRequest& request)
+  class UniqueCURL
   {
-    CURL* curl;
+  protected:
+    std::unique_ptr<CURL, void (*)(CURL*)> p;
+
+  public:
+    UniqueCURL() : p(curl_easy_init(), [](auto x) { curl_easy_cleanup(x); })
+    {
+      if (p.get())
```

`if (p.get())` or
`if (!p.get())`?

---

Definitely the latter!

---

Which brings up the question: how did this ever work, and do we even have a test that positively checks that snapshot download does work and we're not silently failing over to re-sync. And the answer seems to be no!